### PR TITLE
feat: Release rp235x-hal v0.3.0

### PR DESCRIPTION
**Pull request by winksaville**
Prepare to release pr235x-hal version 0.3.0

### CODE REVIEW

**Review by 9names — CHANGES_REQUESTED** [review]
You'll need to update the HAL version for the examples too, cargo won't let you compile things from path if the version specified isn't correct.
https://github.com/rp-rs/rp-hal/blob/e87e2acc233a9e3b54a9287dc2a7285db1969cad/rp235x-hal-examples/Cargo.toml#L35

File: rp235x-hal/CHANGELOG.md [hunk 1]
```diff
@@ -5,9 +5,36 @@ All notable changes to this project will be documented in this file.
...
 
-## Unreleased
+## [Unreleased]
+
+## [0.3.0] - 2024-11-03
 
 ### Changed
 
-- First version
+- Release rp235x-hal v0.3.0
```
Comment (9names) [inline rp235x-hal/CHANGELOG.md:14]: Don't need a comment for the release itself

**Comment by 9names** [comment]
If you look at https://github.com/rp-rs/rp-hal/blob/main/rp2040-hal/CHANGELOG.md, we've tried to include the PR # and the user who made the PR as well.
These are put into the github release notes, to ensure that folks are credited for their work.
An example of that is here: https://github.com/rp-rs/rp-hal/releases/tag/v0.10.2
If you have time to do those things it would be great. Thanks!

File: rp235x-hal/README.md [hunk 2]
```diff
@@ -70,6 +70,16 @@ To include this crate in your project, amend your `Cargo.toml` file to include
...
 
+Or to include this version specifically:
+
+```toml
+rp235x-hal = "0.3.0"
+```
+
+Then, in your Rust code, include the crate like so:
+
+```rust
```
Comment (rursprung) [inline rp235x-hal/README.md:81]: there seems to be something missing here?

**Comment by winksaville** [comment]
> You'll need to update the HAL version for the examples too, cargo won't let you compile things from path if the version specified isn't correct.
> 
> https://github.com/rp-rs/rp-hal/blob/e87e2acc233a9e3b54a9287dc2a7285db1969cad/rp235x-hal-examples/Cargo.toml#L35

Fixed after changing Cargo.toml:
```
$ cargo build
    Updating crates.io index
     Locking 176 packages to latest compatible versions
      Adding ascii-canvas v3.0.0 (latest: v4.0.0)
      Adding bare-metal v0.2.5 (latest: v1.0.0)

..

   Compiling pio-parser v0.2.2
   Compiling rp235x-hal v0.3.0 (/home/wink/prgs/rpi-pico/myrepos/rp-hal/rp235x-hal)
   Compiling pio-proc v0.2.2
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 17.20s
wink@3900x 24-11-04T21:58:15.831Z:~/prgs/rpi-pico/myrepos/rp-hal/rp235x-hal-examples (Release-rp235x-hal-v0.3.0)
```

File: rp235x-hal/README.md (hunk 2 above)
Comment (winksaville) [reply rp235x-hal/README.md:81]: Removed

**Comment by winksaville** [comment]
> If you look at https://github.com/rp-rs/rp-hal/blob/main/rp2040-hal/CHANGELOG.md, we've tried to include the PR # and the user who made the PR as well. These are put into the github release notes, to ensure that folks are credited for their work. An example of that is here: https://github.com/rp-rs/rp-hal/releases/tag/v0.10.2 If you have time to do those things it would be great. Thanks!

Is there a `git log --format=xxx` to get the "PR number" and "github/name" or some other way to generate that info?

**Comment by 9names** [comment]
I usually look at the closed PR list, match against merge commit messages
https://github.com/rp-rs/rp-hal/pulls?q=is%3Apr+is%3Aclosed

**Comment by winksaville** [comment]
I've somehow messed this up, I'm closing it and starting over, sorry!

**Pull request closed without merging**